js/main.js

```diff
@@ -27,6 +27,110 @@ function submitHandler(event) {
+
+  $liElement.appendChild($divColImg);
+  $divColImg.appendChild($img4Entry);
+
+  var $divColEntriesInfo = document.createElement('div');
+  $divColEntriesInfo.className = 'column-half entries-info';
+
+  $liElement.appendChild($divColEntriesInfo);
+
+  var $h2EntryElement = document.createElement('h2');
+  var $entryTitle = document.createTextNode(entries.titleValue);
+  $h2EntryElement.appendChild($entryTitle);
+  $divColEntriesInfo.appendChild($h2EntryElement);
+
+  var $pEntryElement = document.createElement('p');
+  var $pEntryText = document.createTextNode(entries.notesValue);
+
+  $pEntryElement.appendChild($pEntryText);
+  $divColEntriesInfo.appendChild($pEntryElement);
+
+  return $liElement;
+}
+
+var $ulEntries = document.querySelector('#entries-list');
+
+function domContentLoadedHandle(event) {
+
+  for (var i = 0; i < data.entries.length; i++) {
+    var $EntriesList = generateEntriesDOMTree(data.entries[i]);
+    $ulEntries.appendChild($EntriesList);
+  }
+  dataView(data.view);
+}
+
+window.addEventListener('DOMContentLoaded', domContentLoadedHandle);
+
+var $view = document.querySelectorAll('.view');
+var $entriesAnchor = document.querySelector('.entries-anchor');
+var $noEntries = document.querySelector('.no-entries');
+var $newButton = document.querySelector('.new-button');
+
+function viewSwapEntries(event) {
+  event.preventDefault();
+  if (data.entries.length === 0) {
+    $noEntries.classList.remove('hidden');
+  }
+  var $dataViewEntries = event.target.getAttribute('data-view');
+  for (var i = 0; i < $view.length; i++) {
+
+    if ($dataViewEntries === $view[i].getAttribute('data-view')) {
+      $view[i].classList.remove('hidden');
+      dataView($dataViewEntries);
+    } else {
+      $view[i].classList.add('hidden');
+    }
+  }
+  var $viewEntryForm = event.target.getAttribute('data-view');
+  for (i = 0; i < $view.length; i++) {
+
+    if ($viewEntryForm === $view[i].getAttribute('data-view')) {
+      $view[i].classList.remove('hidden');
+      dataView($viewEntryForm);
+    } else {
+      $view[i].classList.remove('hidden');
+      $view[i].classList.add('hidden');
+      $entriesAnchor.className = 'entries-anchor view';
+    }
+  }
+
+}
+$newButton.addEventListener('click', viewSwapEntries);
+$entriesAnchor.addEventListener('click', viewSwapEntries);
+
+function dataView(string) {
+  data.view = string;
+  if (data.entries.length !== 0) {
+    $noEntries.classList.add('hidden');
+  }
+  for (var i = 0; i < $view.length; i++) {
+    if ($view[i].getAttribute('data-view') === string && data.view === 'entries') {
+      $view[i].classList.remove('hidden');
+    } else {
+      $view[i].classList.add('hidden');
+    }
+  }
+  for (i = 0; i < $view.length; i++) {
+    if ($view[i].getAttribute('data-view') === string && data.view === 'entry-form') {
+      $view[i].classList.remove('hidden');
+    } else {
+      $entriesAnchor.className = 'entries-anchor view';
+    }
+  }
```

Comment:
You can reduce all this to the following:
```
  for (var i = 0; i < $view.length; i++) {
    if ($view[i].getAttribute('data-view') === string) {
      $view[i].classList.remove('hidden');
    } else {
      $view[i].classList.add('hidden');
    }
  }
```

Reply:
Examine this and make sure you understand how it works, please ask me any questions if you have them.